src/System.Management.Automation/utils/PsUtils.cs

```diff
@@ -624,6 +624,63 @@ internal static object[] Base64ToArgsConverter(string base64)
         }
     }
 
+    /// <summary>
+    /// A simple implementation of CRC32.
+    /// See "CRC-32 algorithm" in https://en.wikipedia.org/wiki/Cyclic_redundancy_check.
+    /// </summary>
+    internal class CRC32Hash
+    {
+        // CRC-32C polynomial representations
+        private const uint polynomial = 0x1EDC6F41;
+        private static uint[] table;
+
+        static CRC32Hash()
```

Comment:
Could we use pre-defined table and exclude this calculations at startup?

Reply:
Well, the calculated table is the same for the given const `polynomial`, so I guess this could be an option.
@TravisEz13 can you please weigh in?

Reply:
This might be mute if we change to the hash algo that Jason suggested. As we discussed offline about CRC, we should test the algorithm with some representative paths and verify we get appropriate hashes if we do switch. 

I don't think that pre-calculating the table would hurt anything.